### PR TITLE
adding utils unit tests

### LINKED ISSUE #206: Create utils unit tests
## Which area this feature is related to?

/area test-automation

## Issue details

In order to improve the test coverage and have a more reliable testing, we should create unit tests dedicated to [pkg/utils](https://github.com/redhat-developer/alizer/tree/main/go/pkg/utils).

**Target Date**: 01 June 2023

### PR DESCRIPTION
**Pull request by mike-hoang**
## What does this PR do?

adds unit tests for [pkg/utils](https://github.com/redhat-developer/alizer/tree/main/go/pkg/utils)

### Which issue(s) does this PR fix

fixes #206 

### PR acceptance criteria

Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

- [x] Unit/Functional tests

  

- [ ] Documentation

### CODE REVIEW

File: go/pkg/utils/detector_test.go [hunk 1]
```diff
@@ -0,0 +1,1108 @@
...
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GetAnyApplicationFilePathExactMatch(tt.root, tt.propsFiles)
+			if result != tt.expectedResult {
+				t.Errorf("Expected result %s, got %s", tt.expectedResult, result)
+			}
+		})
+	}
+}
+
```
Comment (thepetk) [inline go/pkg/utils/detector_test.go:888]: Shouldn't we add a test for `readAnyApplicationFile` too?

**Review by thepetk — CHANGES_REQUESTED** [review]
I'd say we might need a test for cache.go -> `GetCachedFilePathsFromRoot`
And also for the `language_file_handler.go`

File: go/pkg/utils/detector_test.go (hunk 1 above)
Comment (mike-hoang) [reply go/pkg/utils/detector_test.go:888]: Will add in next commit along with cache.go and language_file_handler.go tests 👍

**Review by thepetk — APPROVED** [review]
/lgtm

**Pull request merged**